flink-yarn-tests/src/main/java/org/apache/flink/yarn/UtilsTest.java

```diff
@@ -97,6 +98,18 @@ public void tooMuchCutoff() {
 		Assert.assertEquals(0, Utils.calculateHeapSize(4000, conf));
 	}
 
+	@Test
+	public void testGetEnvironmentVariables() {
+		Configuration testConf = new Configuration();
+		testConf.setString("yarn.application-master.env.LD_LIBRARY_PATH", "/usr/lib/native");
+
+		Map<String, String> res = Utils.getEnvironmentVariables("yarn.application-master.env.", testConf);
+
+		Assert.assertEquals(1, res.size());
+		Map.Entry<String, String> entry = res.entrySet().iterator().next();
+		Assert.assertEquals("LD_LIBRARY_PATH", entry.getKey());
+		Assert.assertEquals("/usr/lib/native", entry.getValue());
+	}
```

Comment:
I think its okay to assume that the method will work for all prefix strings.
This test is only about the method for extracting the variables.